Additional documentation guidance for alignment with Apple style. (#312)
swiftlang/swift-testing: Documentation/StyleGuide.md

given additional documentation (the documentation in the protocol declaration is
generally sufficient.)

When writing an abstract for a symbol, start the abstract with either a noun or
a verb ending in "s" depending on what kind of symbol it is:
|   Noun               |   Verb ending in *s*      |
|----------------------|---------------------------|
| Associated type      | Enumerations |
| Class                | Function and function macro |
| Constant             | Initializer |
| Enumerated types     | Macro |
| Property             | Method |
| Protocol             | Subscript |
| Structure            |  |
| Type alias           |  |
| Variable             |  |

For instance, when writing the abstract for a class `Order`, you could write:
> An object that stores the details for a specific order from a vendor.

> Or when writing the abstract for an enumeration `Flavor`, you could write:

Reviewer: ```suggestion
Or, when writing the abstract for an enumeration `Flavor`, you could write:
```

> Describes the flavors of an ingredient.

To organize symbols under types, place them in topic groups organized by usage.

> Begin topic group headings insides types with a noun or noun phrase.

Reviewer 2: ```suggestion
Begin topic group headings inside types with a noun or noun phrase.
```

> Begin topic group headings insides types with a nound or noun phrase.

Reviewer 2: ```suggestion
Begin topic group headings insides types with a noun or noun phrase.
```

given additional documentation (the documentation in the protocol declaration is
generally sufficient.)

> 

Reviewer 2: ```suggestion
```

When writing an abstract for a symbol, start the abstract with either a noun or

> a verb ending in s depending on what kind of symbol it is:

Reviewer 2: ```suggestion
a verb ending in "s" depending on what kind of symbol it is:
```

|   Noun               |   Verb ending in *s*      |
|----------------------|---------------------------|
| Associated type      | Enumerations |
| Class                | Function and function macro |
| Constant             | Initializer |
| Enumerated types     | Macro |
| Property             | Method |
| Protocol             | Subscript |
| Structure            |  |
| Type alias           |  |
| Variable             |  |

For instance, when writing the abstract for a class `Order`, you could write:

> <blockquote>

Reviewer 2: Don't use `<blockquote>` or other HTML here. Instead, you can use `> ` to indicate that a line is block-quoted:

```markdown
> An object that stores the details for a specific order from a vendor.
```

> A object that stores the details for a specific order from a vendor.

Reviewer 2: Typo: "A" -> "An"

> A object that stores the details for a specific order from a vendor.

Reviewer 2: ```suggestion
An object that stores the details for a specific order from a vendor.
```

</blockquote>

Or when writing the abstract for an enumeration `Flavor`, you could write:

> <blockquote>

Reviewer 2: Again, use `> ` rather than `<blockquote>`.

Describes the flavors of an an ingredient.
</blockquote>

> 

Reviewer 2: ```suggestion
```

> terms or colloquialisms such as "let's."

Reviewer: ```suggestion
terms or colloquialisms such as _we_, _our_, or _let's_.
```

When writing specific instructions that a reader must follow exactly in order to

> accomplish a task, use the second person ("you", "your", etc.).

Reviewer: ```suggestion
accomplish a task, use the second person (_you_, _your_, and so on).
```

#### Articles

When writing articles or curating content, keep the structure simple and
relatively flat. Place articles in topic groups alongside the symbols.

Some basic rules to follow when creating an article are:

> - Begin an article title with a _gerund_ (a verb ending in "ing").

Reviewer: ```suggestion
- Begin an article title with a _gerund_ (a verb ending in _-ing_).
```

- After the title, include a single sentence that begins with a verb and quickly
describes what the article covers.
- Include an overview to serve as an introduction to the article. If required,
include any setup or configuration tasks as a part of your overview.
- Start section headings with an imperative verb.
- Always follow a section heading with some text to setup the code or problem
you want to solve.

> - Match the names of articles and the files that contain them using kebab-case.

Reviewer: ```suggestion
- The filename for your article should match the title of the article except in lowercase characters, ignoring punctuation, and replacing spaces with hyphens (kebab-case). For example, if the title of your article is _Adding tags to tests_, the filename would be `adding-tags-to-tests.md`.
```

#### API collections

To organize related subsets of symbols, articles, and other content, use an API
collection.

> Some basic rules to follow when creating an API collections are:

Reviewer 2: ```suggestion
Some basic rules to follow when creating an API collection are:
```

- Begin the collection title with a noun that describes what the items in the
collection have in common.
- After the title, include a single sentence that describes the items in the
collection.
- Optionally, include an overview to the collection.
- Organize the symbols under topic group headings. Begin a topic group heading
with a gerund.

> - Match the names of collections and the files that contain them using kebab-case.

Reviewer: ```suggestion
- The filename for your API collection should match the title of the collection page, except in lowercase characters, ignoring punctuation, and replacing spaces with hyphens (kebab-case). For example, if the title of the collection page is _Event tags_, the filename would be `event-tags.md`.
```

Reviewer: Should there be a separate section on file-naming?

Reviewer 2: To consolidate the instructions for articles and collections? Maybe. We'd need to state how to spell filenames for other kinds of documentation _and_ for source files.

> - Match the names of a collections and the files that contain them using kebab-case.

Reviewer 2: ```suggestion
- Match the names of collections and the files that contain them using kebab-case.
```

terms or colloquialisms such as "let's."

When writing specific instructions that a reader must follow exactly in order to
accomplish a task, use the second person ("you", "your", etc.).

#### Articles

When writing articles or curating content, keep the structure simple and
relatively flat. Place articles in topic groups alongside the symbols.

Some basic rules to follow when creating an article are:

> - Begin an article title with a *gerund* (a verb ending in "ing").

Reviewer 2: ```suggestion
- Begin an article title with a _gerund_ (a verb ending in "ing").
```

Some Markdown renderers render `*foo*` as **foo** instead of _foo_.
